Assets/UI/worldinput_CQUI.lua

```diff
@@ -246,6 +246,23 @@ function DefaultKeyUpHandler( uiKey:number )
         cquiHandledKey = true;
     end
 
+    if action["SPREAD_RELIGION"] and CQUI_UnitSupportsSpreadingReligion(unitType) then
+        UnitManager.RequestOperation(selectedUnit, GameInfo.UnitOperations["UNITOPERATION_SPREAD_RELIGION"].Hash);
+        cquiHandledKey = true;
+    end
+
+    if action["REMOVE_HERESY"] and unitType == "UNIT_INQUISITOR" then
+        UnitManager.RequestOperation(selectedUnit, GameInfo.UnitOperations["UNITOPERATION_REMOVE_HERESY"].Hash);
+        cquiHandledKey = true;
+    end
+
+    if (action["RELIGIOUS_HEAL"]
+        and
+        (unitType == "UNIT_MISSIONARY" or unitType == "UNIT_APOSTLE")) then
+        UnitManager.RequestOperation(selectedUnit, GameInfo.UnitOperations["UNITOPERATION_REST_REPAIR"].Hash);
+        cquiHandledKey = true;
+    end
```

Comment:
Religious Heal is the operation gurus can do where they can heal religious units. It happens instantly not over a turn. Did you want to add a shortcut key for that action? Then the check would have to change and code would be
`
UnitManager.RequestOperation(selectedUnit, GameInfo.UnitOperations["UNITOPERATION_RELIGIOUS_HEAL"].Hash);
`

If the action you perform with the code is the intended action then I would recommend changing the name from religious heal to something else to avoid confusion. Also I would recommend changing the check from a unitType based one to a action based one I mention for the other actions.
`
local bCanRestHeal = selectedUnit and UnitManager.CanStartOperation(selectedUnit, GameInfo.UnitOperations["UNITOPERATION_REST_REPAIR"].Hash, nil, true);
`

Reply:
indeed, I so rarely use "religious heal by a guru" that I mistook "heal a religious unit at a holy site" for it. will fix this.

Reply:
done. this is now a generic "H" hotkey for any unit that can rest and heal. not just religious.